docs/core/diagnostics/distributed-tracing-instrumentation-walkthroughs.md

```diff
@@ -0,0 +1,456 @@
+Activity.ParentId:    00-9d5aa439e0df7e49b4abff8d2d5329a9-f16529d0b7c49e44-01
+Activity.DisplayName: StepTwo
+Activity.Kind:        Internal
+Activity.StartTime:   2021-03-18T10:40:51.9441095Z
+Activity.Duration:    00:00:01.0052729
+Resource associated with Activity:
+    service.name: MySample
+    service.instance.id: e0a8c12c-249d-4bdd-8180-8931b9b6e8d0
+
+Activity.Id:          00-9d5aa439e0df7e49b4abff8d2d5329a9-f16529d0b7c49e44-01
+Activity.DisplayName: SomeWork
+Activity.Kind:        Internal
+Activity.StartTime:   2021-03-18T10:40:51.4256627Z
+Activity.Duration:    00:00:01.5286408
+Activity.TagObjects:
+    foo: banana
+    bar: 8
+    otel.status_code: ERROR
+    otel.status_description: Use this text give more information about the error
+Activity.Events:
+    Part way there [3/18/2021 10:40:51 AM +00:00]
+    Done now [3/18/2021 10:40:52 AM +00:00]
+Resource associated with Activity:
+    service.name: MySample
+    service.instance.id: e0a8c12c-249d-4bdd-8180-8931b9b6e8d0
+
+Example work done
+```
+
+Notice that both StepOne and StepTwo include a ParentId that refers to SomeWork. The console is
+not a great visualization of nested trees of work, but many GUI viewers such as
+[Zipkin](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.Zipkin/README.md)
+can show this as a Gantt chart:
+
+[![Zipkin Gantt chart](media/zipkin-nested-activities.jpg)](media/zipkin-nested-activities.jpg)
+
+### Optional: ActivityKind
+
+Activities have an <xref:System.Diagnostics.Activity.Kind%2A?displayProperty=nameWithType> property which
+describes the relationship between the Activity, its parent and its children. By default all new Activities are
+set to <xref:System.Diagnostics.ActivityKind.Internal> which is appropriate for Activities that are an
+internal operation within an application with no remote parent or children. Other kinds can be set using the
+kind parameter on
+<xref:System.Diagnostics.ActivitySource.StartActivity%2A?displayProperty=nameWithType>. See
+<xref:System.Diagnostics.ActivityKind?displayProperty=nameWithType> for other options.
+
+### Optional: Links
+
+When work occurs in batch processing systems a single Activity might represent work on behalf of many
+different requests simultaneously, each of which has its own trace-id. Although Activity is restricted
+to have a single parent, it can link to additional trace-ids using
+<xref:System.Diagnostics.ActivityLink?displayProperty=nameWithType>. Each ActivityLink is
+populated with an <xref:System.Diagnostics.ActivityContext> that
+stores ID information about the Activity being linked to. ActivityContext can be retrieved from in-process
+Activity objects using <xref:System.Diagnostics.Activity.Context%2A?displayProperty=nameWithType> or
+it can be parsed from serialized id information using
+<xref:System.Diagnostics.ActivityContext.Parse(System.String,System.String)?displayProperty=nameWithType>.
```

Comment:
add a note saying - ActivityLinks can only be provided at the source.StartActivityTime, and cannot be added later.
(After reading AddActivityEvent API above, its can be mistakenly assumed that links can also be added later..)